### PR TITLE
Add `LiteralToPrimitiveDeep` Type

### PR DESCRIPTION
**Pull request by alvis**
This PR introduces a new utility type called `LiteralToPrimitiveDeep`, which aims to simplify the handling of data from a json file.

For example, if you want to pass a tsconfig to a function expecting a TsConfigJSON type, then you have to either import the expected type and cast the data to it, or simply cast the data to any.

```
/* function.ts */
import type { TsConfigJson } from 'type-fest';

export function doSomethingWithTSConfig(config: LiteralToPrimitiveDeep<TsConfigJson>) { ... }

/* another.ts */
import tsconfig from 'path/to/tsconfig.json';

// no casting is needed to pass the type check
doSomethingWithTSConfig(tsconfig);

// if LiteralToPrimitiveDeep is not used, you need to cast the imported data like this: 
doSomethingWithTSConfig(tsconfig as TsConfigJson);
doSomethingWithTSConfig(tsconfig as any);
 ```

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -95,19 +95,19 @@ PR welcome for additional commonly needed types and docs improvements. Read the
 npm install type-fest
 ```
 
-*Requires TypeScript >=4.7*
+_Requires TypeScript >=4.7_
```
Comment (sindresorhus) [inline readme.md:98]: Don't make unrelated changes.

File: source/literal-to-primitive-deep.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import type {LiteralToPrimitive} from './literal-to-primitive';
+import type {OmitIndexSignature} from './omit-index-signature';
+
+/**
+ * Given a constant object, return a new object with the same keys but with all the values converted to primitives.
+ * @see LiteralToPrimitive
+ *
```
Comment (sindresorhus) [inline source/literal-to-primitive-deep.d.ts:7]: Follow the existing style with not `*` prefixes.

**Comment by sindresorhus** [comment]
Thanks for contributing. Make sure you adhere to https://github.com/sindresorhus/type-fest/blob/main/.github/contributing.md

**Comment by alvis** [comment]
Sorry for missed the code style. They are corrected.

File: readme.md [hunk 3]
```diff
@@ -157,6 +157,7 @@ Click the type names for complete docs.
 - [`ConditionalExcept`](source/conditional-except.d.ts) - Like `Omit` except it removes properties from a shape where the values extend the given `Condition` type.
 - [`UnionToIntersection`](source/union-to-intersection.d.ts) - Convert a union type to an intersection type.
 - [`LiteralToPrimitive`](source/literal-to-primitive.d.ts) - Convert a [literal type](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types) to the [primitive type](source/primitive.d.ts) it belongs to.
+- [`LiteralToPrimitiveDeep`](source/literal-to-primitive-deep.d.ts) - Like `LiteralToPrimitive` except that it converts the literal types deeply.
```
Comment (sindresorhus) [inline readme.md:160]: It only works with object/array, so it should make that clear, so it should maybe make that clear.

File: source/literal-to-primitive-deep.d.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import type {LiteralToPrimitive} from './literal-to-primitive';
+import type {OmitIndexSignature} from './omit-index-signature';
+
+/**
+Given a constant object, return a new object with the same keys but with all the values converted to primitives.
```
Comment (sindresorhus) [inline source/literal-to-primitive-deep.d.ts:5]: The first line should be the same as the readme description.

File: source/literal-to-primitive-deep.d.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+@see LiteralToPrimitive
+
+Use-case: Deal with data that is imported from a JSON file.
+
+@example
+```
+import type { LiteralToPrimitiveDeep, TsConfigJson } from 'type-fest';
```
Comment (sindresorhus) [inline source/literal-to-primitive-deep.d.ts:12]:
```suggestion
import type {LiteralToPrimitiveDeep, TsConfigJson} from 'type-fest';
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by alfiomy** [comment]
تمام

**Review by alfiomy — APPROVED** [review]

**Comment by alvis** [comment]
@sindresorhus Documentation should be much clearer now :)

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**